client/src/reducers/exerciseEvents.js

```diff
@@ -1,19 +1,21 @@
 import initialState from '../store/initialState';
 import {
   EXERCISE_EVENT_CREATED,
-  USER_LOGIN_SUCCESS,
   EXERCISE_EVENT_DELETED,
-  EXERCISE_EVENT_UPDATED
+  EXERCISE_EVENT_UPDATED,
+  USER_LOGIN_SUCCESS,
 } from '../actions';
 
 export default function exerciseEventsReducer(state = initialState.exerciseEvents, action) {
   switch (action.type) {
     case EXERCISE_EVENT_CREATED:
       return [...state, action.data];
     case EXERCISE_EVENT_UPDATED:
-      const i = state.findIndex(e => e._id === action.data._id);
-      state.splice(i, 1, action.data);
-      return state;
+      return state.map(e =>
```

Comment:
variable name `e` not verbose enough. please rename to ExerciseEventController.java